### PR TITLE
Kajam 11.5.0 schedule

### PR DESCRIPTION
**Pull request by tan-tan-kanarek**

### CODE REVIEW

File: plugins/schedule/base/lib/api/errors/KalturaScheduleErrors.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+/**
+ * @package plugins.schedule
+ * @subpackage api.errors
+ */
+class KalturaScheduleErrors extends KalturaErrors
+{
+	const INVALID_SCHEDULE_END_BEFORE_START = "INVALID_SCHEDULE_END_BEFORE_START;START,END;End time [@END@] must be after start time [@START@]";
+	const MAX_SCHEDULE_DURATION_REACHED = "MAX_SCHEDULE_DURATION_REACHED;MAX;Maximum schedule duration [@MAX@] reached";
+	const MAX_SCHEDULE_DURATION_MUST_MATCH_END_TIME = "MAX_SCHEDULE_DURATION_MUST_MATCH_END_TIME;;Duration must match end time, keep one of them null to calcualte automatically";
```
Comment (MosheMaorKaltura) [inline plugins/schedule/base/lib/api/errors/KalturaScheduleErrors.php]: Typo calcualte

File: plugins/schedule/base/lib/api/KalturaScheduleEvent.php [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+		
+		parent::doFromObject($srcObj, $responseProfile);
+	}
+	
+	/* (non-PHPdoc)
+	 * @see IApiObjectFactory::getInstance($sourceObject, KalturaDetachedResponseProfile $responseProfile)
+	 */
+	public static function getInstance($sourceObject, KalturaDetachedResponseProfile $responseProfile = null)
+	{
+		$object = null;
```
Comment (MosheMaorKaltura) [inline plugins/schedule/base/lib/api/KalturaScheduleEvent.php]: Indentation

File: plugins/schedule/base/lib/api/filters/KalturaScheduleEventFilter.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		if($type)
+		{
+			$c->add(ScheduleEventPeer::TYPE, $type);
+		}
+
+		$filter = $this->toObject();
+		$filter->attachToCriteria($c);
+		$pager->attachToCriteria($c);
+			
+		$list = ScheduleEventPeer::doSelect($c);
```
Comment (MosheMaorKaltura) [inline plugins/schedule/base/lib/api/filters/KalturaScheduleEventFilter.php]: This piece of code, returns many time.
We can use it once by defining a base class that overloads the doSelect and handles the pager, or by calling a static function of peerUtils.
Eventually what interests us is the total count and the list.

File: plugins/schedule/base/lib/model/ScheduleEvent.php [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+		{
+			$periodStart = time();
+		}
+		if(!$periodEnd)
+		{
+			$periodEnd = strtotime('+1 year', $periodStart);
+		}
+		if(!$limit)
+		{
+			$limit = 1000;
```
Comment (MosheMaorKaltura) [inline plugins/schedule/base/lib/model/ScheduleEvent.php]: Can't you get it from - SchedulePlugin::getScheduleEventmaxRecurrences()

File: plugins/schedule/base/lib/model/om/BaseScheduleEventPeer.php [hunk 5]
```diff
@@ -0,0 +1,1122 @@
...
+	/**
+	 * The returned Class will contain objects of the default type or
+	 * objects that inherit from the default.
+	 *
+	 * @param      array $row PropelPDO result row.
+	 * @param      int $colnum Column to examine for OM class information (first is 0).
+	 * @throws     PropelException Any exceptions caught during processing will be
+	 *		 rethrown wrapped into a PropelException.
+	 */
+	public static function getOMClass($row, $colnum)
```
Comment (MosheMaorKaltura) [inline plugins/schedule/base/lib/model/om/BaseScheduleEventPeer.php]: typo - colnum, meant column?

File: plugins/schedule/base/lib/api/filters/KalturaScheduleEventFilter.php (hunk 3 above)
Comment (tan-tan-kanarek) [inline plugins/schedule/base/lib/api/filters/KalturaScheduleEventFilter.php]: I agree, we can create a doFilter method that returns array($list, $totalCount) but it's a project of its own.

**Pull request closed without merging**